### PR TITLE
Done

### PR DESCRIPTION
**Pull request by MaksimusCRS**
[DEMO LINK](https://MaksimusCRS.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

**Review by pasha-melnyk — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
+    const goods = await callback();
+
+    this.setState({ goods });
+  };
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+  render() {
+    const { getAll, get5First, getRedGoods } = goodsAPI;
```
Comment (Elena-brv) [inline src/App.tsx:26]: remove it

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,58 @@
...
+type State = {
+  goods: Good[],
+};
+
+export class App extends React.Component<Props, State> {
+  state: State = {
+    goods: [],
+  };
+
+  loadGoods = async (callback: () => Promise<Good[]>) => {
```
Comment (Elena-brv) [inline src/App.tsx:19]:
```suggestion
  loadGoods = async (getGoods: () => Promise<Good[]>) => {
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,12 +1,58 @@
...
+          </button>
+          <button
+            type="button"
+            className="buttons__button"
+            onClick={() => this.loadGoods(getRedGoods)}
+          >
+            Load red goods
+          </button>
+        </div>
+        {!!this.state.goods.length && <GoodList goods={this.state.goods} />}
```
Comment (Elena-brv) [inline src/App.tsx:54]: do state destructuring

File: src/api/goods.ts [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
     .then(response => response.json());
-}
+};
 
-export const get5First = () => {};
+export const get5First = async () => {
+  const goods = await getAll();
 
-export const getRedGoods = () => {};
+  return goods.sort((good1, good2) => good1.name.localeCompare(good2.name)).slice(0, 5);
```
Comment (Elena-brv) [inline src/api/goods.ts:12]:
```suggestion
  return goods
    .sort((good1, good2) => good1.name.localeCompare(good2.name))
    .slice(0, 5);
```

File: src/components/GoodList/GoodList.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.good-list{
+  list-style: none;
+}
```
Comment (Elena-brv) [inline src/components/GoodList/GoodList.scss:3]: add empty line

File: src/components/GoodList/GoodList.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import './GoodList.scss';
+
+type Props = {
+  goods: Good[],
+};
+
+export const GoodList: React.FC<Props> = ({ goods }) => {
+  return (
```
Comment (Elena-brv) [inline src/components/GoodList/GoodList.tsx:10]: avoid return

**Review by Elena-brv — CHANGES_REQUESTED** [review]

File: src/App.scss [hunk 7]
```diff
@@ -1 +1,27 @@
...
+    background-color: orange;
+    border-radius: 5px;
+    height: 50px;
+    width: 200px;
+
+    &:focus {
+      background-color: yellow;
+    }
+  }
+}
```
Comment (Elena-brv) [inline src/App.scss:27]: empty line

**Review by Elena-brv — APPROVED** [review]

**Pull request closed without merging**